crates/solver/src/solver/single_order_solver.rs

```diff
@@ -56,15 +58,8 @@ impl SingleOrderSolver {
 #[async_trait::async_trait]
 impl Solver for SingleOrderSolver {
     async fn solve(&self, auction: Auction) -> Result<Vec<Settlement>> {
-        let mut orders = auction.orders.clone();
+        let mut orders = get_prioritized_orders(&auction.orders, &auction.external_prices);
 
-        // Randomize which orders we start with to prevent us getting stuck on bad orders.
```

Comment:
Historically, we used randomization as a way to get "unstuck" if we encounter a bad order (or set of bad orders). One concern with this change is that we lose that protection.

Maybe an alternative would be to use a "price-viability" weighted average (so things that are better than, or close to, being price viable get picked almost at the same rate, but the less and less price viable you become, your change of getting picked approaches 0).

Don't know if this is feasible or not, but I do think we should try and introduce randomization one way or the other.

Reply:
I tried to address your suggestion in [987b428](https://github.com/cowprotocol/services/pull/762/commits/987b42881814660735a858577bc0c7af90993ff6). With the simple sorting we knew what we would be getting but with the new weighted sampling we might have to revisit this implementation and adjust weights if it doesn't perform how we want it to.

Reply:
I was expecting it to be much more complicated. Looks nice!